hack/app_sre_create_image_catalog.sh

```diff
@@ -22,9 +22,13 @@ git clone \
 # remove any versions more recent than deployed hash
 REMOVED_VERSIONS=""
 if [[ "$REMOVE_UNDEPLOYED" == true ]]; then
+    # this will need to be modified if at any point we move to a canary like deployment
+    # where not all environments are deployed with the same operator version
     DEPLOYED_HASH=$(
-        curl -s "https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-aws-account-operator.yaml" | \
-            docker run --rm -i quay.io/app-sre/yq -r '.resourceTemplates[]|select(.name="aws-account-operator").targets[]|select(.namespace["$ref"]=="/services/osd-operators/namespaces/aws-account-operator-production.yml")|.ref'
+        curl -s -H "Authorization: Basic $(echo ${APP_INTERFACE_USERNAME}:${APP_INTERFACE_PASSWORD} | base64)" \
+            -g "https://${APP_INTERFACE_BASE_URL}/graphql?query={saas_files:saas_files_v1{name,resourceTemplates{name,targets{namespace{environment{name,labels}},ref}}}}" | \
```

Comment:
I guess these variables are magically set by appsre:
- `APP_INTERFACE_USERNAME`
- `APP_INTERFACE_PASSWORD`
- `APP_INTERFACE_BASE_URL`
and, for that matter
- `APP_SRE_BOT_PUSH_TOKEN`

If I wanted to test this locally, how would I go about it?

Reply:
you would submit a request to get credentials: https://gitlab.cee.redhat.com/service/app-interface#querying-the-app-interface

Reply:
✓